server/main.go

```diff
@@ -126,28 +148,46 @@ func handleMsg() {
 		// Log each message to stdout.
 		t := time.Now().Format(time.ANSIC)
 		if msg.Type == "info" {
+			// Set the username received from the client to the clientInfo present in activeClients.
+			clientInfo := activeClients[msg.ID]
+			clientInfo.Username = msg.Username
+			activeClients[msg.ID] = clientInfo
+
 			color.Green("%s >> %s %s (ID: %s)\n", t, msg.Username, msg.Text, msg.ID)
-		} else if msg.Type == "docReq" {
-			color.Green("%s >> docReq sent from ID: %s\n", t, msg.ID)
 		} else if msg.Type == "operation" {
 			color.Green("operation >> %+v from ID=%s\n", msg.Operation, msg.ID)
 		} else {
 			color.Green("%s >> %+v\n", t, msg)
 		}
 
 		// Broadcast to all active clients.
-		for client, UUID := range activeClients {
+		for id, clientInfo := range activeClients {
 			// Check the UUID to prevent sending messages to their origin.
-			if UUID != msg.ID {
+			if id != msg.ID {
 				// Write JSON message.
-				color.Magenta("writing message to: %s, msg: %+v\n", UUID, msg)
-				err := client.WriteJSON(msg)
+				color.Magenta("writing message to: %s, msg: %+v\n", id, msg)
+				err := clientInfo.Conn.WriteJSON(msg)
 				if err != nil {
 					color.Red("Error sending message to client: %v\n", err)
-					client.Close()
-					delete(activeClients, client)
+					clientInfo.Conn.Close()
+					delete(activeClients, id)
 				}
 			}
 		}
 	}
 }
+
+func handleSync() {
+	for {
+		// Receive document response.
+		docRespMsg := <-docChan
+		color.Cyan("got docRespMsg: %+v", docRespMsg)
+
+		for UUID, clientInfo := range activeClients {
+			if UUID != docRespMsg.ID {
```

Comment:
The bug was fixed by fixing this check, earlier it was `==`. This doesn't send the document response to the client which had requested it.